src/test/java/cz/fi/muni/pa165/Model/Entity/CarTest.java

```diff
@@ -0,0 +1,86 @@
+package cz.fi.muni.pa165.Model.Entity;
+
+import cz.fi.muni.pa165.Model.CarState;
+import org.testng.annotations.Test;
+
+import java.util.Date;
+
+import static org.testng.Assert.*;
+
+/**
+ * @author rtrembecky
+ */
+public class CarTest {
+
+    private Car create() {
+        Date created = new Date();
+        Car car = new Car("R2D2", "456", "Manufacturer", "H510Q", 5, created);
+
+        assertEquals(car.getSerialNumber(), "R2D2");
+        assertEquals(car.getRegPlateNumber(), "456");
+        assertEquals(car.getManufacturer(), "Manufacturer");
+        assertEquals(car.getType(), "H510Q");
+        assertEquals(car.getNumberOfSeats(), 5);
+        assertEquals(car.getEstablishDate(), created);
+
+        return car;
+    }
+
+    @Test
+    public void testChangeState() throws Exception {
+        Car car = create();
+
+        car.changeState(CarState.OK);
+        assertEquals(car.getCarState(), CarState.OK);
+
+        car.changeState(CarState.DISCARDED);
+        assertEquals(car.getCarState(), CarState.DISCARDED);
+
+        car.changeState(CarState.SERVICING);
```

Comment:
I guess we don't want to cover the case when somebody "missclicks" and discards a car by accident and then wants to fix his mistake, do we? Seems like a possible use case, what do you think @jkuchar ?